packages/firestore/src/api/database.ts

```diff
@@ -374,7 +375,8 @@ export class Firestore implements firestore.Firestore, FirebaseService {
     },
     // Exposed via INTERNAL for use in tests.
     disableNetwork: () => this._firestoreClient.disableNetwork(),
-    enableNetwork: () => this._firestoreClient.enableNetwork()
+    enableNetwork: () => this._firestoreClient.enableNetwork(),
+    queue: this._queue
```

Comment:
I'm not a big fan of exposing the whole queue externally since:

1) Although not documented or supported, this is now exposed to consumers of the SDK and could be abused.
2) The only reason this works is because our minification sucks right now and we only minify things that begin or end with `_`.  Really, the entire AsyncQueue class should be minified and all of the methods should have obfuscated names.
3) In general, I'd rather we make the testable API explicit and minimal. Exposing more than we need to leads to a slippery slope heading towards undesirable tight coupling between tests and implementation.

So I'd recommend exposing drainAsyncQueue() or similar (FWIW if you wanted you could make drain() always execute delayed tasks; I don't think it needs to be optional) and either remove the delayedOperationsCount assertions from tests or have the exposed method return the count of operations that were drained or something.

Reply:
I went with `drainAsyncTasks`. In the integration tests that use Firestore, I can indeed always execute the delayed tasks. I can't do this in the spec tests (since that always ends up closing the stream and confuses the tests).

The stream tests use Datastore directly and I allowed the queue to be passed in. That way, I can verify that I have at least one pending task.